srml/staking/src/lib.rs

```diff
@@ -1419,16 +1420,16 @@ impl<T: Trait> Module<T> {
 }
 
 impl<T: Trait> session::OnSessionEnding<T::AccountId> for Module<T> {
-	fn on_session_ending(_ending: SessionIndex, start_session: SessionIndex) -> Option<Vec<T::AccountId>> {
-		Self::new_session(start_session - 1).map(|(new, _old)| new)
+	fn on_session_ending(ending: SessionIndex, _: SessionIndex) -> Option<Vec<T::AccountId>> {
+		Self::new_session(ending + 1).map(|(new, _old)| new)
 	}
 }
 
 impl<T: Trait> OnSessionEnding<T::AccountId, Exposure<T::AccountId, BalanceOf<T>>> for Module<T> {
```

Comment:
```suggestion
impl<T: Trait> historical::OnSessionEnding<T::AccountId, Exposure<T::AccountId, BalanceOf<T>>> for Module<T> {
```

Less confusing. + no need to be imported individually up top.